### PR TITLE
simplify syncNodeImages func

### PR DESCRIPTION
**Pull request by hantmac**
### Ⅰ. Describe what this PR does
Simplify `syncNodeImages` for `ImagePullJob` controller.

### Ⅱ. Does this pull request fix one issue?

NONE

### CODE REVIEW

File: pkg/controller/imagepulljob/imagepulljob_controller.go [hunk 1]
```diff
@@ -304,15 +279,10 @@ func (r *ReconcileImagePullJob) syncNodeImages(job *appsv1alpha1.ImagePullJob, n
 				if tagSpec.Tag != imageTag {
 					continue
 				}
-				if containsObjectRef(tagSpec.OwnerReferences, *ownerRef) {
+				if utilimagejob.FormatNodeImageTagSpecIfHas(tagSpec, imageTag, ownerRef, now) {
```
Comment (FillZpp) [inline pkg/controller/imagepulljob/imagepulljob_controller.go:282]: I prefer the old logic that update tagSpec here, which can be easily compared with the not-found initialization below. `FormatNodeImageTagSpecIfHas` is a little confusing.

Moving containsObjectRef to util is fine.

Comment (hantmac) [reply pkg/controller/imagepulljob/imagepulljob_controller.go:282]: Agree.

File: pkg/util/imagejob/imagejob_reader.go [hunk 2]
```diff
@@ -275,3 +275,15 @@ func SortStatusImageTags(imageStatus *appsv1alpha1.ImageStatus) {
 		return imageStatus.Tags[i].Tag <= imageStatus.Tags[j].Tag
 	})
 }
+
+func FormatNodeImageTagSpecIfHas(tagSpec *appsv1alpha1.ImageTagSpec, imageTag string, ownerRef *v1.ObjectReference, nowTime metav1.Time) bool {
```
Comment (FillZpp) [inline pkg/util/imagejob/imagejob_reader.go:279]: So this can be removed.

File: apis/apps/pub/zz_generated.deepcopy.go [hunk 3]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
Comment (FillZpp) [inline apis/apps/pub/zz_generated.deepcopy.go:1]: I'm not sure how you add the duplicated line in generated files... maybe have to remove them.

Comment (hantmac) [reply apis/apps/pub/zz_generated.deepcopy.go]: I run `make test` and auto generate this line, don't know why 🤔.

**Review by FillZpp — APPROVED** [review]
/lgtm

**Pull request merged**